app/controllers/grade_entry_forms_controller.rb

```diff
@@ -415,6 +416,7 @@ def csv_upload
           end
         end
       end
+      GradeEntryItem.import grades.flatten
```

Comment:
You should move this to right after Line 392 (you want this to be before the block that creates the grades).